WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m

```diff
@@ -606,48 +618,123 @@ - (UITableViewCell *)configureFeaturedImageCellForIndexPath:(NSIndexPath *)index
+        _postGeoLocationCell.tag = PostSettingsRowGeolocationMap;
+    }
+    Coordinate *coordinate = self.post.geolocation;
+    NSString *address = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
+    if (coordinate) {
         CLLocation *postLocation = [[CLLocation alloc] initWithLatitude:coordinate.latitude longitude:coordinate.longitude];
-        NSString *address;
-        if ([[LocationService sharedService] hasAddressForLocation:postLocation]) {
-            address = [LocationService sharedService].lastGeocodedAddress;
+        if ([self.locationService hasAddressForLocation:postLocation]) {
+            address = self.locationService.lastGeocodedAddress;
         } else {
             address = NSLocalizedString(@"Looking up address...", @"Used with posts that are geo-tagged. Let's the user know the the app is looking up the address for the coordinates tagging the post.");
-            [[LocationService sharedService] getAddressForLocation:postLocation
+            __weak __typeof__(self) weakSelf = self;
+            [self.locationService getAddressForLocation:postLocation
                                                         completion:^(CLLocation *location, NSString *address, NSError *error) {
-                                                            [self.tableView reloadData];
+                                                            [weakSelf.tableView reloadData];
                                                         }];
-
+            
         }
-        [geoCell setCoordinate:self.post.geolocation andAddress:address];
-        cell = geoCell;
-        cell.tag = PostSettingsRowGeolocationMap;
+    }
+    [_postGeoLocationCell setCoordinate:coordinate andAddress:address];
+    return _postGeoLocationCell;
+}
+
+- (WPTableViewCell *)geoLocationActivityCell {
+    if (!_geoLocationActivityCell) {
+        _geoLocationActivityCell = [[WPTableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:nil];
+        _geoLocationActivityCell.accessoryType = UITableViewCellAccessoryNone;
+        [WPStyleGuide configureTableViewCell:_geoLocationActivityCell];
+        UIActivityIndicatorView *spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+        _geoLocationActivityCell.accessoryView = spinner;
+        [spinner startAnimating];
+        _geoLocationActivityCell.textLabel.text = NSLocalizedString(@"Finding your location...", @"Geo-tagging posts, status message when geolocation is found.");
+    }
+    return _geoLocationActivityCell;
+}
 
+- (UITableViewCell *)configureGeolocationCellForIndexPath:(NSIndexPath *)indexPath
+{
+    WPTableViewCell *cell;
+    switch (indexPath.row) {
+        case (0): {
+            return self.geoLocationSwitchCell;
+        }
+        break;
```

Comment:
For both case statements,
No need for the `(` `)`s on the case statement.
Not sure if this is in the style guide or not, but the `{` `}` are unnecessary unless there are variables to define in block scope.  
Also, since the case is returning, the `break;` is unneeded.